src/tron/TronEngine.ts

```diff
@@ -1242,6 +1244,55 @@ export class TronEngine extends CurrencyEngine<TronTools, SafeTronWalletInfo> {
     return edgeTransaction
   }
 
+  async makeWithdrawExpireUnfreezeTransaction(): Promise<EdgeTransaction> {
+    const contractJson = {
+      parameter: {
+        value: {
+          owner_address: base58ToHexAddress(this.walletLocalData.publicKey)
+        }
+      },
+      type: 'WithdrawExpireUnfreezeContract'
+    }
+
+    const txOtherParams: TxBuilderParams = { contractJson }
+    const { transactionHex } = await this.txBuilder(txOtherParams)
+    const networkFee = await this.calcTxFee({ unsignedTxHex: transactionHex })
+
+    const claimedAmount = this.stakingStatus.stakedAmounts.reduce(
+      (sum, stakedAmount) => {
+        const { nativeAmount, otherParams, unlockDate } = stakedAmount
+        if (
+          unlockDate != null &&
+          new Date(unlockDate) > new Date() &&
+          otherParams?.type === 'WITHDRAWEXPIREUNFREEZE'
+        ) {
+          return add(sum, nativeAmount)
+        }
+        return sum
+      },
+      '0'
+    )
+
+    const edgeTransaction: EdgeTransaction = {
+      txid: '',
+      date: 0,
+      currencyCode: this.currencyInfo.currencyCode,
+      blockHeight: 0,
+      nativeAmount: sub(claimedAmount, networkFee),
+      isSend: false,
```

Comment:
This should be true, since it's a transaction we ourselves have constructed. `false` is for stuff we detect off the network.

Reply:
I marked it false since the net effect is positive to the wallet (staked amount returning to spendable balance minus fee). There's no other incoming transaction to show the funds have been added back to the wallet

Reply:
I see. That means the bug is in the GUI, since the "isSend" concept should be independent of whether money was added or removed from the balance. This is getting into refactor territory, so let's merge as-is.